### DISCUSSION TITLE
Redirect log output for commands other than "server"

### DISCUSSION THREAD

**Discussion by JamesBoon** [dropwizard/dropwizard#8602] [General]
Is there any way to redirect _all_ log output to stdout instead of a configured file appender if I start any other [command](https://www.dropwizard.io/en/stable/manual/core.html#commands) than `server` e.g. when running `check`? 

I [saw the option](https://www.dropwizard.io/en/stable/manual/core.html#configuration) to override `config.yml` settings by passing system properties. 
Changing a single appender works with something like this:
`-Ddw.logging.appenders[0].type=console -Ddw.logging.appenders[0].logFormat="%msg%n"`
But this only changes the first appender in the array. How would it look like if you want to replace the whole `appenders` array with a single appender? How can I remove any other appender?

Else I thought maybe a [FilterFactory](https://www.dropwizard.io/en/stable/manual/core.html#logging-filters) would help, but it does not work with other commands than `server`.  

The only other way I see here is to have a different `config.yml` for my CLI commands - but I'd really like to use the same file.

Am I missing something? Or is this a missing feature?

**Comment by zUniQueX** [comment]
Hi @JamesBoon. The config overrides only accept simple string values and arrays of strings. Hence, they cannot be used to override the whole appenders object.

The filter factories also work for other commands since they reside in `dropwizard-logging` and don't depend on any logic related to the server command. The docs only provide an example for a `FilterFactory<IAccessEvent>` that is indeed only related to request logging within the server command. To obtain a filter factory for a "normal" appender rather than a request log appender, you'll need to provide a `FilterFactory<ILoggingEvent>`.

I think the simplest solution for your use case is to use [environment variables](https://www.dropwizard.io/en/stable/manual/core.html#environment-variables) for the logging levels of the appenders. Then you can provide an environment variable with the value `OFF` to disable logging for the other appenders.

**Comment by JamesBoon** [comment]
Thank you @zUniQueX, that was the hint I needed!

If anyone else needs it:
```yaml
appenders:
  - type: console
    threshold: ${DW_LOG_THRESHOLD_CONSOLE:-OFF}
    # ...
  - type: file
    threshold: ${DW_LOG_THRESHOLD_FILE:-INFO}
    # ...
```

This way, the console logger is `OFF` by default, but can be switched on by setting the environment variable `DW_LOG_THRESHOLD_CONSOLE` to any log level like e.g. `INFO`.  
The file logger is on by default and can be switched off by setting `DW_LOG_THRESHOLD_FILE=OFF`.

Just before anybody asks, yes the same thing can be achieved via passing system properties. But this should only be used if the _number and order_ of the log appenders is _always_ fixed!
`-Ddw.logging.appenders[0].threshold=INFO -Ddw.logging.appenders[1].threshold=OFF`
